### DISCUSSION TITLE
Are there free pins available for logic sensing?

### DISCUSSION THREAD

**Discussion by mterry63** [CircuitSetup/Expandable-6-Channel-ESP32-Energy-Meter#124] [General]
Hello,

I plan to use my 6 input board for monitoring current for commercial power as well as my generator during emergency power use. I was wondering if there are any free pins available on the ESP32 I could use to sense power for each input (commercial/generator). I was considering 2 120V relays with SPST contacts as the "sense" input devices on each input. I could easily wire this to a separate ESP, but if there are pins already available, less hardware and IP addresses consumed. 

I'm using ESPHome and Home Assistant with the energy meter board, I'm assuming I can just add the logic pins in there if available.

**Comment by CircuitSetup** [comment]
You can definitely use open pins on the ESP32. I would recommend using the left most GPIOs since there is an open header on that side. Assuming you do not have add-on boards using a CS pin, you can use 13, 14, 27, 26, 25

**Comment by mterry63** [comment]
Thanks for your help, I used GPIOs 13 and 14 and tested on my bench by jumpering to ground, works great.
